extensions/google-java/src/main/kotlin/com/connectrpc/extensions/GoogleJavaJSONStrategy.kt

```diff
@@ -33,10 +37,16 @@ class GoogleJavaJSONStrategy : SerializationStrategy {
      * This unchecked cast assumes the underlying class type is
      * a Google GeneratedMessageV3.
      */
-    @Suppress("UNCHECKED_CAST")
```

Comment:
I moved the suppressions down to each relevant statement and added a comment as to why it's actually safe. It wasn't previously safe, but now it is because of an added check that the given type is actually a sub-type of the desired type (which I've changed from `GeneratedMessageV3` to `Message` since that is all that is needed and `GeneratedMessageV3` is overly precise and prevents things like use of dynamic messages).